src/arch/ia32/descriptor/segment.rs

```diff
@@ -1,155 +1,301 @@
+//! A module containing the implementation, tests and revolving types around
+//! the [`SegmentDescriptor`] structure to describe the permissions and
+//! capacities of a memory segment.
 pub use super::Granularity;
-use super::{split_address, split_limit};
 pub use crate::arch::ia32::PrivilegeLevel;
+use bit_field::BitField;
+use configuration::Configuration;
 use core::fmt;
+use permissions::Permissions;
 
-pub mod lower;
-mod upper;
+mod configuration;
+mod permissions;
 
+/// The default operation size modes of a memory segment.
+///
+/// It could be either 16 bits (`0`) or 32 bits (`0`). It performs different
```

Comment:
I believe it was supposed to be `or 32 bits ('1')`

Reply:
Fixed in 141a5a9